### PR TITLE
Remove several unnecessary null checks

### PR DESCRIPTION
**Pull request by megawac**
Where they are handled implicitly by other underscore methods. I'd say this makes the source of a number of these functions quite a bit nicer.

### CODE REVIEW

File: underscore.js [hunk 1]
```diff
@@ -216,9 +213,8 @@
         accumulator = {};
       }
     }
-    if (obj == null) return accumulator;
     iteratee = optimizeCb(iteratee, context, 4);
-    var keys = obj.length !== +obj.length && _.keys(obj),
+    var keys = !isArrayLike(obj) && _.keys(obj),
```
Comment (jdalton) [inline underscore.js]: rogue `isArrayLike`

**Comment by jdalton** [comment]
I can dig it. I kept similar checks in for the perf arms race but if they were removed here I'd likely remove them too.

File: underscore.js (hunk 1 above)
Comment (megawac) [inline underscore.js]: Intentional call, same thing going on in [reduce](https://github.com/jashkenas/underscore/blob/master/underscore.js#L186), map, some and others

Comment (jdalton) [inline underscore.js]: Ah, I thought it was reverted, guess not. No worries.

**Pull request merged**